packages/android_intent/android/src/main/java/dev/flutter/plugins/androidintent/MethodCallHandlerImpl.java

```diff
@@ -0,0 +1,132 @@
+package dev.flutter.plugins.androidintent;
+
+import android.content.ComponentName;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.provider.Settings;
+import android.text.TextUtils;
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
+import io.flutter.plugin.common.MethodChannel.Result;
+import java.util.ArrayList;
+import java.util.Map;
+
+/** Forwards incoming {@link MethodCall}s to {@link IntentSender#send}. */
+public class MethodCallHandlerImpl implements MethodCallHandler {
+  private final IntentSender sender;
+
+  /**
+   * Uses the given {@code sender} for all incoming calls.
+   *
+   * <p>This assumes that the sender's context and activity state are managed elsewhere and
+   * correctly initialized before being sent here.
+   */
+  public MethodCallHandlerImpl(IntentSender sender) {
+    this.sender = sender;
+  }
+
+  /**
+   * Parses the incoming call and forwards it to the cached {@link IntentSender}.
+   *
+   * <p>Always calls {@code result#success}.
+   */
+  @Override
+  public void onMethodCall(MethodCall call, Result result) {
+    String action = convertAction((String) call.argument("action"));
+    Integer flags = call.argument("flags");
+    String category = call.argument("category");
+    Uri data = call.argument("data") != null ? Uri.parse((String) call.argument("data")) : null;
+    Bundle arguments = convertArguments((Map<String, ?>) call.argument("arguments"));
+    String packageName = call.argument("package");
+    ComponentName componentName =
+        (!TextUtils.isEmpty(packageName) && !TextUtils.isEmpty((String) call.argument("component")))
+            ? new ComponentName(packageName, (String) call.argument("component"))
+            : null;
+
+    sender.send(action, flags, category, data, arguments, packageName, componentName);
+
+    result.success(null);
+  }
+
+  private static String convertAction(String action) {
+    switch (action) {
```

Comment:
BTW, not related to the migration, but I've always wondered why we don't make it easy to launch a new `Activity` the way that any Android dev would normally do. It might be nice to at least have a method on the Dart side that simply takes the qualified name of an `Activity` to launch.

Reply:
Reasonable, looks like flutter/flutter#34296 is the existing feature request for it.

It struck me as weird while I was refactoring this that we don't have any way to get the result of the launch either.